### ISSUE TITLE
tach measurement speed config bit

### ISSUE THREAD

**Issue by ProjectPatatoe** [RobTillaart/ADT7470#6]
related:#4
(not neccessarily an issue with the library as much as an important point to note when using the chip)

ok there are a few things Ive found odd about tachometer readings. "Fan Speed Measurement Rate" does nothing as far as I can tell on the surface.

Here's a quick test where fan speed is full (255). In the loop,
 * setFastTach()
 * print register to show it's changed
 * getTach() to print raw Tach reading (pre-formula)  every  250ms

results:
```
16:45:29.758 -> --------fast--------config40: 00100001
16:45:29.758 -> Tach2: 0000011001111100     1660
16:45:29.758 -> Tach2: 0000011001111100     1660
16:45:29.758 -> Tach2: 0000011001111101     1661
16:45:29.758 -> Tach2: 0000011001111100     1660
16:45:29.758 -> Tach2: 0000011001111100     1660
16:45:29.758 -> Tach2: 0000011001111100     1660
16:45:29.758 -> Tach2: 0000011001111110     1662
16:45:29.758 -> Tach2: 0000011010000001     1665
16:45:29.758 -> Tach2: 0000011010000001     1665
16:45:29.758 -> Tach2: 0000011001111111     1663
16:45:29.758 -> --------slow--------config40: 00000001
16:45:29.758 -> Tach2: 0000011001111111     1663
16:45:29.758 -> Tach2: 0000011001111111     1663
16:45:29.758 -> Tach2: 0000011010000000     1664
16:45:29.758 -> Tach2: 0000011010000000     1664
16:45:29.781 -> Tach2: 0000011010000000     1664
16:45:30.048 -> Tach2: 0000011010000001     1665
16:45:30.283 -> Tach2: 0000011010000000     1664
16:45:30.551 -> Tach2: 0000011010000000     1664
16:45:30.786 -> Tach2: 0000011010000001     1665
16:45:31.054 -> Tach2: 0000011010000001     1665
16:45:31.289 -> --------fast--------config40: 00100001
16:45:31.289 -> Tach2: 0000011001111111     1663
16:45:31.558 -> Tach2: 0000011001111101     1661
16:45:31.825 -> Tach2: 0000011001111101     1661
16:45:32.060 -> Tach2: 0000011001111110     1662
16:45:32.327 -> Tach2: 0000011001111111     1663
16:45:32.562 -> Tach2: 0000011010000000     1664
16:45:32.830 -> Tach2: 0000011001111110     1662
16:45:33.065 -> Tach2: 0000011001111101     1661
16:45:33.333 -> Tach2: 0000011001111101     1661
16:45:33.568 -> Tach2: 0000011001111101     1661
16:45:33.838 -> --------slow--------config40: 00000001
16:45:33.838 -> Tach2: 0000011001111101     1661
16:45:34.073 -> Tach2: 0000011001111101     1661
16:45:34.342 -> Tach2: 0000011001111100     1660
16:45:34.577 -> Tach2: 0000011001111011     1659
16:45:34.844 -> Tach2: 0000011001111011     1659
16:45:35.079 -> Tach2: 0000011001111011     1659
16:45:35.345 -> Tach2: 0000011001111010     1658
16:45:35.578 -> Tach2: 0000011001111001     1657
16:45:35.847 -> Tach2: 0000011001111001     1657
16:45:36.081 -> Tach2: 0000011001111011     1659
```
The config change doesn't seem to change anything. What I expect is that 3-4 readings remain the same during "slow" bit as stated on the bottom of pg23.

Same test again **but with a changing fan speed** to ensure that the tach is changing. The fan is speeding up during "fast" and slowing down during "slow"
Results:
```
17:05:40.188 -> --------fast--------config40: 00100001
17:05:40.188 -> Tach2: 0000110010001111     3215
17:05:40.424 -> Tach2: 0000100100101011     2347
17:05:40.693 -> Tach2: 0000011111001001     1993
17:05:40.929 -> Tach2: 0000011100110101     1845
17:05:41.198 -> Tach2: 0000011011101110     1774
17:05:41.434 -> Tach2: 0000011011010000     1744
17:05:41.704 -> Tach2: 0000011011000001     1729
17:05:41.940 -> Tach2: 0000011010111000     1720
17:05:42.209 -> Tach2: 0000011010110100     1716
17:05:42.445 -> Tach2: 0000011010110001     1713
17:05:42.716 -> --------slow--------config40: 00000001
17:05:42.716 -> Tach2: 0000011010110001     1713
17:05:42.951 -> Tach2: 0000011110101001     1961
17:05:43.221 -> Tach2: 0000100010100100     2212
17:05:43.457 -> Tach2: 0000100101111001     2425
17:05:43.727 -> Tach2: 0000101001000010     2626
17:05:43.963 -> Tach2: 0000101011110010     2802
17:05:44.198 -> Tach2: 0000101101101011     2923
17:05:44.469 -> Tach2: 0000101111001100     3020
17:05:44.705 -> Tach2: 0000110000011000     3096
17:05:44.974 -> Tach2: 0000110001001111     3151
17:05:45.210 -> --------fast--------config40: 00100001
17:05:45.210 -> Tach2: 0000110001110001     3185
17:05:45.480 -> Tach2: 0000100100100100     2340
17:05:45.716 -> Tach2: 0000011111000101     1989
17:05:45.986 -> Tach2: 0000011100110010     1842
17:05:46.222 -> Tach2: 0000011011101101     1773
17:05:46.492 -> Tach2: 0000011011001101     1741
17:05:46.727 -> Tach2: 0000011010111110     1726
17:05:46.996 -> Tach2: 0000011010110111     1719
17:05:47.231 -> Tach2: 0000011010110101     1717
17:05:47.499 -> Tach2: 0000011010101111     1711
17:05:47.733 -> --------slow--------config40: 00000001
17:05:47.733 -> Tach2: 0000011010101011     1707
17:05:48.001 -> Tach2: 0000011110010011     1939
17:05:48.236 -> Tach2: 0000100010000110     2182
17:05:48.505 -> Tach2: 0000100101110010     2418
17:05:48.740 -> Tach2: 0000101000110100     2612
17:05:49.009 -> Tach2: 0000101011000101     2757
17:05:49.243 -> Tach2: 0000101101010110     2902
17:05:49.512 -> Tach2: 0000101110110011     2995
17:05:49.747 -> Tach2: 0000101111111000     3064
17:05:50.014 -> Tach2: 0000110000100100     3108
```
The raw tach increases as it slows and decreases as it spins faster as expected (pg23). However, still not seeing the change of where the readings are the same during slow.

My theory:

I believe, doing a tach read resets the timer and immediately begins a new measurement. The usefulness would only be realized when set in automatic mode where the fans are controled by the tach readings with the limits set in the chip ahead of time. this would result in the fans reacting faster or slower to temperature changes.

I feel like theres not much a solution for the simple use of the library / manual control. I don't have the temperature sensors that go with this so I can't test the automatic portion of the chip.

**Comment by RobTillaart** [comment]
@ProjectPatatoe 
Sorry for the late reply, I'm checking all my libs for open issues,

Maybe we should close this issue as I expect no progress because of lack of hardware.

**Comment by ProjectPatatoe** [comment]
I think this might actually be fixed due to #4 . I think that it was reading wrong and so this actually not the issue. I just know that as it is now, it is working as expected. I'll close for now and someone can open a new one if they still have a problem
